crates/turborepo-lib/src/shim.rs

```diff
@@ -330,11 +348,19 @@ impl RepoState {
                 self.spawn_local_turbo(&canonical_local_turbo, shim_args),
             ))
         } else {
-            try_check_for_updates(&shim_args, get_version());
+            let global_version = get_version();
+            try_check_for_updates(&shim_args, global_version);
+            debug!("Running command as global turbo");
+
+            if !turbo_version_in_range(global_version, &self.root) {
+                return Err(anyhow!(
+                    "Project specifies that version is 9, you're running 8."
```

Comment:
What's idomatic for passing `version_req_string` back out so that I can use it here?

I could also make it `try_turbo_version_in_range` which has ok/err and precompose it in the function? Or is there a better pattern?